### PR TITLE
Cslang

### PR DESCRIPTION
**Pull request by benmurphyy**
### Description

Integrate [cslang](https://github.com/source-academy/c-slang) into source academy frontend as another language tab that allows for programming in C.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### CODE REVIEW

File: src/commons/application/ApplicationTypes.ts [hunk 1]
```diff
@@ -279,13 +283,15 @@ export const ALL_LANGUAGES: readonly SALanguage[] = [
...
-  ...pyLanguages
+  ...pyLanguages,
+  ...cLanguages
 ];
 // TODO: Remove this function once logic has been fully migrated
 export const getLanguageConfig = (
   chapter: Chapter,
   variant: Variant = Variant.DEFAULT
 ): SALanguage => {
+  //TODO: come back fix
```
Comment (RichDom2185) [inline src/commons/application/ApplicationTypes.ts:294]: Is this TODO meant to be acted on? Or is this PR ready?

**Review by RichDom2185 — COMMENTED** [review]
Looking forward to the C integration! Could you run `yarn format` to fix the format issues?

Thanks!

File: src/commons/application/ApplicationTypes.ts (hunk 1 above)
Comment (benmurphyy) [reply src/commons/application/ApplicationTypes.ts:294]: Seems to be a TODO carried over from when I first branched from frontend. Will remove!

**Comment by benmurphyy** [comment]
PR is currently a work in progress (in midst of enhancing compilation error reporting on frontend). Will request your review @RichDom2185 after its completed and will fix the formatting issues before that.

**Comment by RichDom2185** [comment]
> PR is currently a work in progress

Ok sure, I'll mark it as draft first in that case so that it won't be touched by others. Thanks!

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**